### PR TITLE
Added log only if signature match

### PR DESCRIPTION
**Pull request by sinarest1608**
Added check if "report_handler" present.

Should be merged before #9

### CODE REVIEW

**Review by mbentz-uf — CHANGES_REQUESTED** [review]
The behavior of the package has changed since migrating from python_db_logger, but the code hasn't been updated to reflect the change in behavior. I didn't notice until reviewing the code further that we can get rid of `add_log_entries`, `build_default_extras`, and simplify the logic to flow be more straightforward. The refactor looks like:

```python
def emit(self, record):
    # Only process if report_handler key is present
    if "report_handler" not in record.__dict__.keys():
        return

    if not hasattr(self, "start_time"):
        self.start_time = datetime.utcnow()

    # Add entry to levelname
    self.add_entry_to_sheet(sheet=record.levelname, entry=_clean_record_msg(record.msg))

    entry, sheet = utils.retrieve_data_and_sheet_name(
        record.__dict__["report_handler"])

    # Add to additional sheet if specified
    if (sheet):
        self.add_entry_to_sheet(sheet=sheet, entry=entry)

def _clean_record_msg(raw_msg: string): 
    cleaned_log_msg = raw_msg
    cleaned_log_msg = cleaned_log_msg.strip()
    cleaned_log_msg = cleaned_l
```

**Review by sinarest1608 — COMMENTED** [review]
@mbentz-uf 
The 
```python
if "report_handler" in record.__dict__.keys():
``` 
was done to accommodate handlers like Kafka and sql in future. New change looks like this:

```python
    def emit(self, record):
        # Only process if report_handler key is present
        # if "report_handler" not in record.__dict__.keys():
        #     return

        if not hasattr(self, "start_time"):
            self.start_time = datetime.utcnow()

        if "report_handler" in record.__dict__.keys():
            # Add entry to levelname
            self.add_entry_to_sheet(sheet=record.levelname,
                                    entry=self._clean_record_msg(record.msg))

            entry, sheet = utils.retrieve_data_and_sheet_name(
                record.__dict__["report_handler"])

            # Add to additional sheet if specified
            if (sheet):
                self.add_entry_to_sheet(sheet=sheet, entry=entry)

    def _clean_record_msg(self, raw_msg: str):
        cleaned_log_msg = raw_msg
        cleaned_log_msg = cleaned_log_msg.strip()
        return cleaned_log_msg
```
Another change which had to be done to accommodate the above was replacing 
```python
headers, values = itemgetter("headers", "values")(row_data)
```
 by 
```python
headers, values = utils.get_headers_and_content(row_data)
``` 
in `write_report` function because write report now supports adding logs using an array of objects.

Should I proceed and commit this change?

**Comment by mbentz-uf** [comment]
> was done to accommodate handlers like Kafka and sql in future
I'm not sure what you mean, that is what checking for "report_handler" at the top of the emit function does. Unless that key exists then we do nothing in the emit function and return, otherwise we continue.

**Comment by sinarest1608** [comment]
I think I misunderstood your comment. I had the picture in mind that we plan to integrate other handlers in this project itself. 
I will add the suggested changes and push to the branch.

**Comment by mbentz-uf** [comment]
@sinarest1608 Can you include the changes from PR #9 in this PR? It should be easier to test that way.

**Comment by sinarest1608** [comment]
@mbentz-uf 
Should I add all the tests here itself and discard the PR #9 branch?

**Comment by mbentz-uf** [comment]
Yes that would be great!

**Comment by sinarest1608** [comment]
@mbentz-uf Added the fixed tests.

**Comment by mbentz-uf** [comment]
The latest commit wouldn't show on the PR until I changed the base branch to main and back to develop.

**Review by mbentz-uf — CHANGES_REQUESTED** [review]
Separate unit test into multiple tests to cover each test case.

**Comment by sinarest1608** [comment]
@mbentz-uf 
I have separated the tests into different functions.

File: tests/report_handler/test_report_handler.py [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+        self.run_test_cases()
+
+        # Reading the generated reporting and adding assertion checks
+        for file in os.listdir("test_logs"):
+            excel = pd.read_excel("test_logs/"+file, None)
+
+            # # Checks if two sheets generated
+            # sheets = len(excel.keys())
+            # self.assertEqual(2, sheets)
+
```
Comment (mbentz-uf) [inline tests/report_handler/test_report_handler.py:81]: Can you remove these comments?

Comment (sinarest1608) [reply tests/report_handler/test_report_handler.py]: @mbentz-uf 
Done!

**Comment by mbentz-uf** [comment]
Can you take a look at the unit tests? They are failing.

**Comment by sinarest1608** [comment]
@mbentz-uf 
All the tests work perfectly fine for me when I run them in VsCode or from the command line. Seems like automated workflows are failing due to missing optional packages. The log file for every test function is deleted after the function is completed, which might not be happening due to the optional dependencies resulting in extra elements. 
```python
======================================================================
FAIL: test_log_file (tests.report_handler.test_report_handler.TestReportHandler)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/runner/work/report_handler/report_handler/tests/report_handler/test_report_handler.py", line 113, in test_log_file
    self.assertEqual(log_sequence_expected, log_sequence)
AssertionError: Lists differ: ['INFO', 'DEBUG', 'WARNING', 'ERROR'] != ['DEBUG', 'DEBUG', 'DEBUG', 'DEBUG', 'INFO', 'DEBUG', 'WARNING', 'ERROR']

First differing element 0:
'INFO'
'DEBUG'

Second list contains 4 additional elements.
First extra element 4:
'INFO'

- ['INFO', 'DEBUG', 'WARNING', 'ERROR']
+ ['DEBUG', 'DEBUG', 'DEBUG', 'DEBUG', 'INFO', 'DEBUG', 'WARNING', 'ERROR']

----------------------------------------------------------------------
```
I will try adding them to `pyproject.toml` and check if they work.

**Comment by sinarest1608** [comment]
@mbentz-uf 
That worked!

**Pull request merged**